### PR TITLE
Added a refactoring to separate sending file logic from StaticRoute dispatcher

### PR DESCRIPTION
**Pull request by dkuznetsov**
## What these changes does?

It separates file sending logic from StaticRoute dispatcher.

## How to test your changes?

Functionally, there's no change from the user's perspective, but I've modified the unit tests to handle the change properly.

## Related issue number

It should address the following question:
https://github.com/KeepSafe/aiohttp/issues/839

## Checklist

- [ ] Code is written and well
- [ ] Tests for the changes are provided
- [ ] Documentation reflects the changes

### CODE REVIEW

**Comment by asvetlov** [comment]
Please resolve merge conflicts

File: aiohttp/helpers.py [hunk 1]
```diff
@@ -519,3 +519,131 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+        if n < count:
+            loop.add_writer(out_fd, self._sendfile_cb, fut, out_fd, in_fd,
+                            offset + n, count - n, loop, True)
+        else:
+            fut.set_result(None)
+
+    @asyncio.coroutine
+    def _sendfile_system(self, req, resp, fobj, count):
+        """
+        Write `count` bytes of `fobj` to `resp` starting from `offset` using
```
Comment (amirouche) [inline aiohttp/helpers.py:567]: There is not offset argument.

File: aiohttp/helpers.py [hunk 2]
```diff
@@ -519,3 +519,131 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+        the ``sendfile`` system call.
+
+        `req` should be a :obj:`aiohttp.web.Request` instance.
+
+        `resp` should be a :obj:`aiohttp.web.StreamResponse` instance.
+
+        `fobj` should be an open file object.
+
+        `offset` should be an integer >= 0.
```
Comment (amirouche) [inline aiohttp/helpers.py:576]: There is no offset argument.

**Comment by dkuznetsov** [comment]
There's a problem during Python3.4 test runs with this one.

**Comment by asvetlov** [comment]
I know. You've missed `yield from` in `FileSender.send()`.
I've fixed it.

**Pull request merged**